src/main/java/com/celarli/commons/vfs/provider/google/GCSFileObject.java

```diff
@@ -128,6 +128,12 @@ protected FileType doGetType() throws Exception {
                 return FileType.FOLDER;
             }
             else {
+
+                //blobs are not listed if url starts with /.
+                if (url.startsWith("/")) {
```

Comment:
@adhudiya 

what happens if user defined root location? (like / )

Reply:
in that situation if block will be executed not the else

Reply:
why are we removing leading slash ?

Reply:
internally it doesn't list blobs if leading slash is there, you will find same code on line number 107